docs/howto/migrate-plugins/charm-to-uv.rst

```diff
@@ -0,0 +1,107 @@
+.. _howto-migrate-to-uv:
+
+Migrate from the Charm plugin to the uv plugin
+==============================================
+
+For charms that use `uv`_, Charmcraft has a :ref:`craft_parts_uv_plugin`. Migrating
+from the Charm plugin provides some benefits, such as using uv during the build
+process not having to maintain a separate ``requirements.txt`` file. If the
+charm to be migrated does not currently use uv, refer to the
+`uv documentation <https://docs.astral.sh/uv/guides/projects/>`_ for instructions
+on how to use uv for a Python project.
+
+Update the project file
+-----------------------
+
+The first step is to update the project file to include the correct parts definition.
+Depending on the history of a specific charm, it may not have an explicitly-included
+``parts`` section determining how to build the charm. In this case, a ``parts`` section
+can be created as follows:
+
+.. code-block:: yaml
+    :caption: charmcraft.yaml
+
+    parts:
+      my-charm:  # This can be named anything you want
+        plugin: uv
+        source: .
+
+Add optional dependency groups
+------------------------------
+
+If the charm has `dependency groups`_ that should be included when creating the virtual
+environment, the ``uv-groups`` key can be used to include those groups when creating
+the virtual environment.
+
+.. note::
+    This is useful and encouraged, though not mandatory, for keeping track of
+    library dependencies, as covered in the next section.
+
+Include charm library dependencies
+----------------------------------
+
+Unlike the Charm plugin, the uv plugin does not install the dependencies for
+included charmlibs. If any of the charm libraries used have ``PYDEPS``, these will
+need to be added to the charm's dependencies, potentially as their own
+`dependency group <dependency groups_>`_.
```

Comment:
We linked to this quite recently – generally not necessary to do it again so soon.

Reply:
Kept this link and removed the other one because I rearranged the sections a bit.